cranelift/codegen/src/isa/x64/abi.rs

```diff
@@ -202,1095 +713,25 @@ fn is_callee_save_baldrdash(r: RealReg) -> bool {
-                    Amode::imm_reg(off as u32, regs::rsp()),
-                    from_reg,
-                    ty,
-                ))
-            }
-        }
-    }
-
-    fn emit_copy_retval_to_reg<C: LowerCtx<I = Self::I>>(
-        &self,
-        ctx: &mut C,
-        idx: usize,
-        into_reg: Writable<Reg>,
-    ) {
-        match &self.sig.rets[idx] {
-            &ABIArg::Reg(reg, ty, _) => ctx.emit(Inst::gen_move(into_reg, reg.to_reg(), ty)),
-            &ABIArg::Stack(off, ty, _) => {
-                let ret_area_base = self.sig.stack_arg_space;
-                let sp_offset = off + ret_area_base;
-                // TODO handle offsets bigger than u32::max
-                debug_assert!(sp_offset >= 0);
-                debug_assert!(sp_offset <= u32::max_value() as i64);
-                ctx.emit(load_stack(
-                    Amode::imm_reg(sp_offset as u32, regs::rsp()),
-                    into_reg,
-                    ty,
-                ));
-            }
-        }
-    }
-
-    fn emit_call<C: LowerCtx<I = Self::I>>(&mut self, ctx: &mut C) {
-        let (uses, defs) = (
-            mem::replace(&mut self.uses, Default::default()),
-            mem::replace(&mut self.defs, Default::default()),
-        );
-
-        if let Some(i) = self.sig.stack_ret_arg {
-            let dst = ctx.alloc_tmp(RegClass::I64, types::I64);
-            let ret_area_base = self.sig.stack_arg_space;
-            debug_assert!(
-                ret_area_base <= u32::max_value() as i64,
-                "large offset for ret area NYI"
-            );
-            ctx.emit(Inst::lea(
-                Amode::imm_reg(ret_area_base as u32, regs::rsp()),
-                dst,
-            ));
-            self.emit_copy_reg_to_arg(ctx, i, dst.to_reg());
-        }
-
-        match &self.dest {
-            &CallDest::ExtName(ref name, RelocDistance::Near) => ctx.emit_safepoint(
-                Inst::call_known(name.clone(), uses, defs, self.loc, self.opcode),
-            ),
-            &CallDest::ExtName(ref name, RelocDistance::Far) => {
-                let tmp = ctx.alloc_tmp(RegClass::I64, types::I64);
-                ctx.emit(Inst::LoadExtName {
-                    dst: tmp,
-                    name: Box::new(name.clone()),
-                    offset: 0,
-                    srcloc: self.loc,
-                });
-                ctx.emit_safepoint(Inst::call_unknown(
-                    RegMem::reg(tmp.to_reg()),
-                    uses,
-                    defs,
-                    self.loc,
-                    self.opcode,
-                ));
-            }
-            &CallDest::Reg(reg) => ctx.emit_safepoint(Inst::call_unknown(
-                RegMem::reg(reg),
-                uses,
-                defs,
-                self.loc,
-                self.opcode,
-            )),
-        }
-    }
+    // Sort for determinism.
+    regs.sort_by_key(|r| r.to_reg().get_index());
```

Comment:
We can use an unstable sort here, not that it matters...

Reply:
Good point! (I had initially thought we wanted a stable sort to ensure deterministic code output, but of course the registers will be unique, so it doesn't matter...)